<?xml version="1.0"?>
<launch>
  
  <arg name="joystick" default="true"/>
  
  <arg name="bag_file" default="/home/chur/siar_2018-05-09-11-40-21.bag"/>
  <arg name="play_bag" default="true"/>
  
  <arg name="start" default="0"/>
  <arg name="rate" default="1"/>
  
  <param name="use_sim_time" value="true" />
  
  # Bag file
  <node name="bag_player" type="play" pkg="rosbag" args="$(arg bag_file) --clock -s $(arg start) -r $(arg rate)" if="$(arg play_bag)" output="screen"/>
  
  <node pkg="odom_to_tf" type="odom_to_tf.py" name="odom_to_tf"/>
  
  <!-- Representation stuff-->
  <node pkg="base_station" type="base_station" name="bs" >
    
    # Use simulation time fro TFs
    <remap from="/front_web/rgb/camera_info" to="/front/rgb/camera_info"/>
    <remap from="/front_web/rgb/image_raw" to="/front/rgb/image_raw"/>
    <remap from="/front_left_web/rgb/camera_info" to="/front_left/rgb/camera_info"/>
    <remap from="/front_left_web/rgb/image_raw" to="/front_left/rgb/image_raw"/>
    <remap from="/front_right_web/rgb/camera_info" to="/front_right/rgb/camera_info"/>
    <remap from="/front_right_web/rgb/image_raw" to="/front_right/rgb/image_raw"/>
  </node>
  
  <include file="$(find siar_stl_model)/launch/siar.launch"/>
  
  <node name="uncompresser_front_rgb" pkg="image_transport" type="republish"
        args="compressed in:=/front/rgb/image_raw raw out:=/front/rgb/image_raw"/>
   <node name="uncompresser_front" pkg="image_transport" type="republish"
        args="compressedDepth in:=/front/depth_registered/image_raw raw out:=/front/depth_registered/image_raw"/>
   
   
  <node pkg="depth2cloud" type="depth2cloud_node" name="front_depth2cloud_node" output="screen" >
    <remap from="depth" to="/front/depth_registered/image_raw"/>              # Topic for the camera depth
    <remap from="info" to="/front/depth_registered/camera_info"/>   # Topic for the camera info
    <remap from="cloud" to="/front/points"/>                # Topic name for output point cloud
    <param name="range_min" value = "0.5"/>                             # Min range in meters for point cloud generation
    <param name="range_max" value = "6.0"/>                             # Max range in meters for point cloud generation
    <param name="down_sampling" value = "1"/>                   # Camera down sampling factor. Values: 1, 2 or 4. The bigger the faster.
  </node>
  
  <node name="uncompresser_front_left" pkg="image_transport" type="republish"
        args="compressedDepth in:=/front_left/depth_registered/image_raw raw out:=/front_left/depth_registered/image_raw"/>
   <node name="uncompresser_front_left_rgb" pkg="image_transport" type="republish"
        args="compressed in:=/front_left/rgb/image_raw raw out:=/front_left/rgb/image_raw"/>
   
  <node pkg="depth2cloud" type="depth2cloud_node" name="front_left_depth2cloud_node" output="screen" >
    <remap from="depth" to="/front_left/depth_registered/image_raw"/>              # Topic for the camera depth
    <remap from="info" to="/front_left/depth_registered/camera_info"/>   # Topic for the camera info
    <remap from="cloud" to="/front_left/points"/>                # Topic name for output point cloud
    <param name="range_min" value = "0.5"/>                             # Min range in meters for point cloud generation
    <param name="range_max" value = "6.0"/>                             # Max range in meters for point cloud generation
    <param name="down_sampling" value = "1"/>                   # Camera down sampling factor. Values: 1, 2 or 4. The bigger the faster.
  </node>
  
  <node name="uncompresser_front_right" pkg="image_transport" type="republish"
        args="compressedDepth in:=/front_right/depth_registered/image_raw raw out:=/front_right/depth_registered/image_raw"/>
   <node name="uncompresser_front_right_rgb" pkg="image_transport" type="republish"
        args="compressed in:=/front_right/rgb/image_raw raw out:=/front_right/rgb/image_raw"/>
   
  <node pkg="depth2cloud" type="depth2cloud_node" name="front_right_depth2cloud_node" output="screen" >
    <remap from="depth" to="/front_right/depth_registered/image_raw"/>              # Topic for the camera depth
    <remap from="info" to="/front_right/depth_registered/camera_info"/>   # Topic for the camera info
    <remap from="cloud" to="/front_right/points"/>                # Topic name for output point cloud
    <param name="range_min" value = "0.5"/>                             # Min range in meters for point cloud generation
    <param name="range_max" value = "6.0"/>                             # Max range in meters for point cloud generation
    <param name="down_sampling" value = "1"/>                   # Camera down sampling factor. Values: 1, 2 or 4. The bigger the faster.
  </node>
  
  
</launch>
